forum/views.py

```diff
@@ -1,7 +1,89 @@
+# GNU General Public License for more details.
+
 from __future__ import unicode_literals
 
 from django.shortcuts import render
+from helpers.mixins import ValidateSubdomainMixin, RedirectToSignupMixin
+from student.models import Student
+from timetable.models import Semester
+from forum.models import Transcript, Comment
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from serializers import TranscriptSerializer, CommentSerializer
+
+
+class ForumView(ValidateSubdomainMixin, RedirectToSignupMixin, APIView):
+    """ Returns all forums for the user making the request. """
+
+    def get(self, request):
+        student = Student.objects.get(user=request.user)
+        return Response(
+            {'invited_transcripts': TranscriptSerializer(
+                student.invited_transcripts, many=True).data,
+             'owned_transcripts': TranscriptSerializer(
+                 student.owned_transcripts, many=True).data},
+            status=status.HTTP_200_OK)
+
+
+class ForumTranscriptView(ValidateSubdomainMixin, RedirectToSignupMixin, APIView):
+    def get(self, request, sem_name, year):
+        student = Student.objects.get(user=request.user)
+        semester = Semester.objects.get(name=sem_name, year=year)
+        transcript = Transcript.objects.get(owner=student, semester=semester)
+        return Response({'transcript': TranscriptSerializer(transcript).data},
+                        status=status.HTTP_200_OK)
+
+    def post(self, request, sem_name, year):
+        student = Student.objects.get(user=request.user)
+        semester = Semester.objects.get(name=sem_name, year=year)
+        transcript = Transcript.objects.get(owner=student, semester=semester)
+        comment = Comment.objects.create(author=student,
+                                         content=request.data['content'],
+                                         timestamp=request.data['timestamp'],
+                                         transcript=transcript)
+
+        return Response({'comment': CommentSerializer(comment).data},
+                        status=status.HTTP_200_OK)
+
+    def put(self, request, sem_name, year):
+        student = Student.objects.get(user=request.user)
+        semester = Semester.objects.get(name=sem_name, year=year)
+
+        transcript = Transcript.objects.create(
+            owner=student, semester=semester)
+        transcript.save()
+
+        return Response({'transcript': TranscriptSerializer(transcript).data},
+                        status=status.HTTP_200_OK)
+
+    def patch(self, request, sem_name, year):
+        student = Student.objects.get(user=request.user)
+        semester = Semester.objects.get(name=sem_name, year=year)
+
+        transcript = Transcript.objects.get(owner=student, semester=semester)
+        if request.data['action'] == 'add':
+            transcript.advisors.add(Student.get(jhed=request.data['jhed']))
+        else:
```

Comment:
I would make "remove" an explicit action instead of "else" here

Reply:
Does that mean we can just write methods called `add` and `remove` in place of `patch`? I wasn't sure if REST framework method names could be changed.

Reply:
Oh, I think I misunderstood - You want us to use `elif request.data['action'] == 'remove'` instead of `else` correct?